launcher/net/ChecksumValidator.h

```diff
@@ -64,8 +65,10 @@ class ChecksumValidator : public Validator {
 
     auto validate(QNetworkReply&) -> bool override
     {
-        if (m_expected.size() && m_expected != hash()) {
+        if (m_expected.size() && m_expected != hash())
+        {
             qWarning() << "Checksum mismatch, download is bad.";
+            qDebug() << "Expected" << m_expected.toHex() << "but got" << hash().toHex();
```

Comment:
I would combine both of those, so the message would look something like this
instead: "Checksum mismatch, expected %1 but got %2.". Putting it on warning
level would be nice too.